String in map which saves []string isn't working

**V version:** V 0.1.28 bdecbc1.3c432f5 
**OS:** Manjaro

**What did you do?**
```
module main

fn main() {
    mut test_map := map[string][]string

    test_map['test'] = ['hello', 'hi']

    edit_map_with_mut(mut test_map)

    mut test_map2 := map[string][]string

    test_map2['hi'] = [',dfns', 'dfjisd']

    edit_map_with_mut(mut test_map2)
}

fn edit_map_with_mut(mut m map[string][]string) {
    m['hello'] = ['a', 'b']

    if 'test' in m { m['test2'] = ['hi', 'b']}
}
```

**What did you expect to see?**
The program runs succesful
 
**What did you see instead?**
```==================
/home/louis/.cache/v/test.tmp.c: In function ‘main__edit_map_with_mut’:
/home/louis/.cache/v/test.tmp.c:7551:31: error: incompatible type for argument 1 of ‘map_exists’
 7551 |  if (_IN_MAP(tos_lit("test"), m)) {
      |                               ^
      |                               |
      |                               map_string_array_string * {aka struct map *}
/home/louis/.cache/v/test.tmp.c:280:36: note: in definition of macro ‘_IN_MAP’
  280 | #define _IN_MAP(val, m) map_exists(m, val)
      |                                    ^
/home/louis/.cache/v/test.tmp.c:5488:28: note: expected ‘map’ {aka ‘struct map’} but argument is of type ‘map_string_array_string *’ {aka ‘struct map *’}
 5488 | static bool map_exists(map m, string key) {
      |                        ~~~~^
...
==================```

--- a/vlib/builtin/map_test.v
+++ b/vlib/builtin/map_test.v
@@ -301,3 +301,15 @@ fn test_map_keys_to_array() {
 	println(sarr)
 	assert sarr == "['a', 'c']"
 }
+
+fn map_in_mut(mut m map[string]int) {
+	if 'one' in m {
+		m['one'] = 2
+	}
+}
+
+fn test_map_in_mut() {
+	mut m := {'one': 1}
+	map_in_mut(mut m)
+	assert m['one'] == 2
+}

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -1912,6 +1912,9 @@ fn (mut g Gen) infix_expr(node ast.InfixExpr) {
 			g.write('_IN_MAP(')
 			g.expr(node.left)
 			g.write(', ')
+			if node.right_type.is_ptr() {
+				g.write('*')
+			}
 			g.expr(node.right)
 			g.write(')')
 		} else if right_sym.kind == .string {

